### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by AlexYup**

### CODE REVIEW

File: ts-service/package.json [hunk 1]
```diff
@@ -11,12 +11,14 @@
     "author": "simonpiscitelli@gmail.com",
     "private": true,
     "dependencies": {
-        "@yupana/core-lib-poc": "^0.1.0",
-        "typescript": "^3.3.3333",
-        "tslib": "^1.9.3"
+        "@yupana/class-generator": "^0.1.3",
+        "@yupana/framework-service-types": "file:../../yupana-service-lib",
```
Comment (AlexYup) [inline ts-service/package.json]: Should it be a separate package or included into framework-types ?

At the moment it exports only 3 types from https://github.com/YupanaInc/Yupana-Framework/blob/build/3.0.0/server/lib/services/server.js : 
```
export {Task};
export {Service};
export {ServicePackage};
```

Comment (lportales) [reply ts-service/package.json]: Maybe framework-types... ?

File: ts-service/src/index.ts [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
-    const baseMsg = "FROM TS-SERVICE: ";
-
-    console.log(`${baseMsg}${message}`);
-
-    objs.forEach(obj => {
-        console.log(JSON.stringify(obj, null, 2));
-    });
+import {Service} from "@yupana/framework-service-types";
+import {YContext} from "@yupana/framework-types/lib/YContext";
+import {YRole} from "@yupana/framework-types/lib/YRoles";
```
Comment (lportales) [inline ts-service/src/index.ts:2]: I missed in the original PR, why do we have Yroles and YSessions (plural) ?

File: ts-service/src/index.ts [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
+import {YSession} from "@yupana/framework-types/lib/YSessions";
+import {YLog, YQuery} from "@yupana/framework-types";
+import {Model} from "../../simple/ts-classes/model";
+
+async function worker(serv: YContext<Service>, role: YRole, session: YSession) {
+    console.log(">>> worker >>> about to fetch data ...");
+
+    const cnx = await session.dbserver.connect("simple", 'simple');
```
Comment (lportales) [inline ts-service/src/index.ts]: Should we normalize quote types ?

File: ts-service/src/index.ts [hunk 4]
```diff
@@ -1,36 +1,44 @@
...
+    const query = YQuery.From('simple.model')
+                .Select()
+                .Execute();
+    const result = await cnx.find(query);
+    const m:Model = await role.simple.createClass("model");
+    let data = await result.iterator.next();
+    while(data) {
```
Comment (lportales) [inline ts-service/src/index.ts]:
```suggestion
    while (data) {
```

**Review by lportales — COMMENTED** [review]
Out of the scope : can we introduce a tslint file ?

**Comment by AlexYup** [comment]
js/tslint is up to app devs. It would be a good idea to add it to sample invoicing as a guideline.

File: ts-service/package.json (hunk 1 above)
Comment (pcrispim) [reply ts-service/package.json]: i agree — the naming of the npm packages we're publishing aren't... "optimal"...

- the `@yupana/class-generator` does not generate any classes, it's the base data types (`YTClass`, `YTslot`, `YTarray`, etc.)
- there's `@yupana/framework-types` already, which are the interface definitions for the likes of `YSession`, `YQuery`, `YMongoDB`, `YRole` and such — core framework types?

but these `@yupana/framework-service-types` seem to be accurate — they are types that only matter to services, they cannot be used elsewhere...

i suggest dropping the "framework" from the package names — and...
- rename `@yupana/class-generator` to `@yupana/base-types`
- rename `@yupana/framework-types` to `@yupana/core-types`
- and rename this latest `@yupana/framework-service-types` to `@yupana/service-types`
 (not sure how this works from an npm standpoint, must it be a new package or does it support renaming?)

Comment (AlexYup) [reply ts-service/package.json]: Yeah, I like single-purpose packages like `@yupana/service-types` myself. They seem idiomatic to npm. I am just not quite sure about maintenance cost. It depends on core types, so all changes there must be propagated upstream. 
Including service types into the core will significantly simplify updates, and we gonna have a lot of them at first.

Comment (AlexYup) [reply ts-service/package.json]: Added to the core https://github.com/YupanaInc/yupana-core-lib/pull/3

File: ts-service/src/index.ts (hunk 2 above)
Comment (AlexYup) [reply ts-service/src/index.ts:2]: Will update the library in v0.1.5

**Review by pcrispim — APPROVED** [review]

**Pull request merged**